### ISSUE TITLE
BigCommerce Plugin Activation Crashing Fresh WP Install Completely

### ISSUE THREAD

**Issue by stavrosthewonderchicken** [bigcommerce/bigcommerce-for-wordpress#295]
### Expected behavior

Activate plugin, connect WP to BigCommerce, happy days

### Actual behavior

Crashing fresh test WP site completely on activation of plugin.

### Steps to reproduce behavior

1. Install BigCommerce for Wordpress on fresh WP install, no plugins activated, default theme (hosted on WPEngine)
2. Activate plugin. "There has been a critical error on this website. Please check your site admin email inbox for instructions."

Manually deleting the plugin via SFTP returns site to working state.

Here's the last few entries from the PHP error log:

```
[Sat Jul 31 04:21:18.295854 2021] [php7:error] [pid 11073]  [client 180.228.219.36:42098] PHP Fatal error:  Uncaught  BigCommerce\\Exceptions\\Channel_Not_Found_Exception: Primary channel is  not set. in  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/Connections.php:28\nStack  trace:\n#0  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/Connections.php(40):  BigCommerce\\Taxonomies\\Channel\\Connections->primary()\n#1  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/BC_Status.php(71):  BigCommerce\\Taxonomies\\Channel\\Connections->current()\n#2  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/BC_Status.php(41):   BigCommerce\\Taxonomies\\Channel\\BC_Status->get_current_channel_status()\n#3   /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Container/Taxonomies.php(162):  BigCommerce\\Taxonomies\\Channel\\BC_Status->admin_notices()\n#4  /nas/content/live/btpdev1/wp-includes/class-wp-hook.php(303):  BigCommerce\\Container\\Taxonomies->BigCommerce\\Container in  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/Connections.php  on line 28, referer: https://btpdev1.wpengine.com/wp-admin/plugins.php

[Sat Jul 31 04:21:26.082402 2021] [php7:error] [pid 11073]  [client 180.228.219.36:42290] PHP Fatal error:  Uncaught  BigCommerce\\Exceptions\\Channel_Not_Found_Exception: Primary channel is  not set. in  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/Connections.php:28\nStack  trace:\n#0  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/Connections.php(40):  BigCommerce\\Taxonomies\\Channel\\Connections->primary()\n#1  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/BC_Status.php(71):  BigCommerce\\Taxonomies\\Channel\\Connections->current()\n#2  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/BC_Status.php(41):   BigCommerce\\Taxonomies\\Channel\\BC_Status->get_current_channel_status()\n#3   /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Container/Taxonomies.php(162):  BigCommerce\\Taxonomies\\Channel\\BC_Status->admin_notices()\n#4  /nas/content/live/btpdev1/wp-includes/class-wp-hook.php(303):  BigCommerce\\Container\\Taxonomies->BigCommerce\\Container in  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/Connections.php  on line 28, referer:  https://btpdev1.wpengine.com/wp-admin/edit.php?page=bigcommerce_welcome&post_type=bigcommerce_product

 [Sat Jul 31 04:21:30.708762 2021] [php7:error] [pid 11073]  [client 180.228.219.36:42464] PHP Fatal error:  Uncaught  BigCommerce\\Exceptions\\Channel_Not_Found_Exception: Primary channel is  not set. in  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/Connections.php:28\nStack  trace:\n#0  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/Connections.php(40):  BigCommerce\\Taxonomies\\Channel\\Connections->primary()\n#1  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/BC_Status.php(71):  BigCommerce\\Taxonomies\\Channel\\Connections->current()\n#2  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/BC_Status.php(41):   BigCommerce\\Taxonomies\\Channel\\BC_Status->get_current_channel_status()\n#3   /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Container/Taxonomies.php(162):  BigCommerce\\Taxonomies\\Channel\\BC_Status->admin_notices()\n#4  /nas/content/live/btpdev1/wp-includes/class-wp-hook.php(303):  BigCommerce\\Container\\Taxonomies->BigCommerce\\Container in  /nas/content/live/btpdev1/wp-content/plugins/bigcommerce/src/BigCommerce/Taxonomies/Channel/Connections.php  on line 28, referer:  https://btpdev1.wpengine.com/wp-admin/edit.php?post_type=bigcommerce_product
```

**Comment by stavrosthewonderchicken** [comment]
https://wordpress.org/support/topic/channel_not_found_exception-primary-channel-is-not-set/

**Comment by rikere** [comment]
Workaround failed here as well. It locked in an infinite loop about getting things ready after I clicked to login with BC. I didn't even get a login prompt.

![image](https://user-images.githubusercontent.com/22123117/127916209-19abfefa-0390-4055-bc1c-4541b74432ca.png)

The start over button doesn't appear to have any links/events bound on it. Console output has no errors other than a failure to get a favicon (clean/new install).

Does anyone know the URL Path to get to the API Key input option?

**Comment by designbureau** [comment]
Having the same issue.

**Comment by tyla-taylor** [comment]
This has been addressed in v4.15.1.
